### PR TITLE
fs/timerfd/eventfd: using anonymous inodes and using isr to notify poll waiter.

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
1. using anonymous inodes to manage timerfd and eventfd.
2. using isr to notify poll waiter to optimize notification speed and avoid race condition.
## Impact
Remove /var/event/xxx and /var/timer/xxx character file. 
## Testing
daily test.

### CODE REVIEW

File: fs/vfs/fs_eventfd.c [hunk 1]
```diff
@@ -332,7 +311,7 @@ static ssize_t eventfd_do_read(FAR struct file *filep, FAR char *buffer,
 
   /* Device ready for read */
 
-  if (dev->mode_semaphore)
+  if (filep->f_oflags & EFD_SEMAPHORE)
```
Comment (pkarashchenko) [inline fs/vfs/fs_eventfd.c:314]: Optional
```suggestion
  if ((filep->f_oflags & EFD_SEMAPHORE) != 0)
```

File: fs/vfs/fs_eventfd.c [hunk 2]
```diff
@@ -528,13 +505,15 @@ static int eventfd_do_poll(FAR struct file *filep, FAR struct pollfd *fds,
...
-  int ret;
-  int new_fd;
   FAR struct eventfd_priv_s *new_dev;
+  int new_fd;
+  int ret;
 
-  /* devpath: EVENT_FD_VFS_PATH + /efd (4) + %u (10) + null char (1) */
-
-  char devpath[sizeof(CONFIG_EVENT_FD_VFS_PATH) + 4 + 10 + 1];
+  if (flags & ~(EFD_NONBLOCK | EFD_SEMAPHORE | EFD_CLOEXEC))
```
Comment (pkarashchenko) [inline fs/vfs/fs_eventfd.c:512]: Optional
```suggestion
  if ((flags & ~(EFD_NONBLOCK | EFD_SEMAPHORE | EFD_CLOEXEC)) != 0)
```

File: fs/vfs/fs_timerfd.c [hunk 3]
```diff
@@ -434,30 +369,40 @@ static int timerfd_poll(FAR struct file *filep, FAR struct pollfd *fds,
...
+
+  intflags = enter_critical_section();
+
+  /* Increment timer expiration counter */
+
+  dev->counter++;
+
+  /* If this is a repetitive timer, then restart the watchdog */
+
+  if (dev->delay)
```
Comment (pkarashchenko) [inline fs/vfs/fs_timerfd.c:403]: Optional
```suggestion
  if (dev->delay > 0)
```

File: fs/vfs/fs_timerfd.c [hunk 4]
```diff
@@ -518,18 +437,15 @@ int timerfd_create(int clockid, int flags)
 
   /* Sanity checks. */
 
-  if (clockid != CLOCK_REALTIME &&
-      clockid != CLOCK_MONOTONIC &&
-      clockid != CLOCK_BOOTTIME)
+  if ((clockid != CLOCK_REALTIME &&
+       clockid != CLOCK_MONOTONIC &&
+       clockid != CLOCK_BOOTTIME) ||
+      (flags & ~(TFD_NONBLOCK | TFD_CLOEXEC)))
```
Comment (pkarashchenko) [inline fs/vfs/fs_timerfd.c:443]: Optional
```suggestion
      (flags & ~(TFD_NONBLOCK | TFD_CLOEXEC)) != 0)
```

File: fs/vfs/fs_eventfd.c (hunk 1 above)
Comment (Donny9) [reply fs/vfs/fs_eventfd.c]: Done!

File: fs/vfs/fs_eventfd.c (hunk 2 above)
Comment (Donny9) [reply fs/vfs/fs_eventfd.c]: Done!

File: fs/vfs/fs_timerfd.c (hunk 4 above)
Comment (Donny9) [reply fs/vfs/fs_timerfd.c]: Done! thanks

File: fs/vfs/fs_timerfd.c (hunk 3 above)
Comment (Donny9) [reply fs/vfs/fs_timerfd.c]: Done!

**Comment by pkarashchenko** [comment]
LGTM!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**